tests/cpp/test_multidevice_matmul.cpp

```diff
@@ -180,7 +180,7 @@ TEST_F(DistributedMatmulTest, Matmul_LayoutTN_NoComms) {
       executor_cache.getMostRecentKernelRuntime();
   EXPECT_THAT(
       kernel_runtime->fusionSegments()->groups(),
-      Contains(HeuristicIs(SchedulerType::ExprEval)).Times(2));
+      Contains(HeuristicIs(SchedulerType::ExprEval)).Times(1));
```

Comment:
Can you check how much slower test_transformer_forward and _backward https://github.com/NVIDIA/Fuser/blob/34fa83b0afbfab92c3ee35ec6aace8f22d975455/tests/python/multidevice/test_multidevice.py#L1334 become with this PR? I think they'll be affected because qkv-split-reshape-transpose is done without kernels. 

Context: I was hoping this PR to be perf neutral at least. I knew you plan to fix markAliasPrepare so the perf will eventually recover. I'm not sure how deep and how long the dip will be. I also wonder what quick fixes we can throw into markAliasPrepare to avoid the perf dip for just logical split.